torch/_vmap_internals.py

```diff
@@ -266,3 +270,21 @@ def wrapped(*args):
         finally:
             torch._C._vmapmode_decrement_nesting()
     return wrapped
+
+# Warning: Do NOT use, this is a temporary workaround and may be removed.
+# This is a version of vmap that allows None to pass through. It enables us to wrap the call
+# to the autograd engine, in `autograd.grad` which may return None if any of the inputs are
+# unused. See the issue discussing this: https://github.com/facebookresearch/functorch/issues/159.
+def _unsafe_vmap_for_autograd(func: Callable, in_dims: in_dims_t = 0, out_dims: out_dims_t = 0) -> Callable:
+    @functools.wraps(func)
+    def wrapped(*args):
+        _check_out_dims_is_int_or_int_tuple(out_dims, func)
+        vmap_level = torch._C._vmapmode_increment_nesting()
+        try:
+            batched_inputs, batch_size = _create_batched_inputs(in_dims, args, vmap_level, func)
+            batched_outputs = func(*batched_inputs)
+            # Don't validate the outputs
+            return _unwrap_batched(batched_outputs, out_dims, vmap_level, batch_size, func, allow_none_pass_through=True)
+        finally:
+            torch._C._vmapmode_decrement_nesting()
```

Comment:
Can we just combine this with the `_vmap` in this file by adding a `allow_none_pass_through` flag to it and documenting what it's used for? That one is also "private" (I don't remember what uses it)